pyam/utils.py

```diff
@@ -94,13 +97,36 @@ def write_sheet(writer, name, df, index=False):
         writer.sheets[name].set_column(i, i, width)  # assumes xlsxwriter as engine
 
 
+def get_excel_file_with_kwargs(path, **kwargs):
+    """Return a `pandas.ExcelFile` and a dict of unused kwargs.
+
+    When reading an Excel file, this function finds keyword arguments that
+    should be passed to `pandas.ExcelFile`, and returns a `pandas.ExcelFile`
+    instance along with the remaining keyword arguments (which presumably
+    will be used for other purposes by the calling function).
+    """
+    EXCEL_FILE_KWS = ('engine', 'storage_options', 'engine_kwargs')
+    kwargs = kwargs.copy()
+    excel_file_kwargs = {
+        k: kwargs.pop(k) for k in EXCEL_FILE_KWS if k in kwargs
+    }
+    if packaging.version.parse(importlib.metadata.version("pandas")) \
+            < packaging.version.parse("2.2.0"):
```

Comment:
Very nice idea to warn the users, but it's a bit too much to warn all users (where 99% won't even know what engine-kwargs are... Hence a proposal for a more targeted warning.
```suggestion
    # TODO remove when bumping minimum pandas dependency to >= 2.2
    if "engine_kwargs" in excel_file_kwargs and packaging.version.parse(
        importlib.metadata.version("pandas")
    ) < packaging.version.parse("2.2.0"):
```

Reply:
> Very nice idea to warn the users, but it's a bit too much to warn all users (where 99% won't even know what engine-kwargs are)... Hence a proposal for a more targeted warning.

Yes, I totally agree. Thanks!